File: 357/solution.py
from random import random
from math import *

need_check = []
p = [False for i in range(int(1e8))]


def check(a):
    upper = int(sqrt(a) + 1)
    for i in range(2, upper):
        if a % i == 0 and p[i + a // i]:
            return False
    return True


if __name__ == "__main__":
    p[0], p[1] = True, True
    for i in range(2, int(1e8)):
        if not p[i]:
            need_check.append(i - 1)
            for j in range(i << 1, int(1e8), i):
                p[j] = True
    sum = 0
    for i in need_check:
        if check(i):
            sum += i
    print(sum)

File: 357/solution.cpp
#include <iostream>/*{{{*/
#include <map>
#include <math.h>
#include <set>
#include <vector>
const int e4  = 10000;
const int e8  = e4 * e4;
const int mod = 1e9 + 7;
using namespace std; /*}}}*/
vector<int> need_check;
bool p[e8];
bool check(int a) {
    for (int i = 2; i * i <= a; i++) {
        if (a % i == 0 && p[i + a / i])
            return false;
    }
    return true;
}
int main() {
    p[0] = p[1] = true;
    for (int i = 2; i < e8; i++)
        if (!p[i]) {
            need_check.push_back(i - 1);
            for (int j = i << 1; j < e8; j += i)
                p[j] = true;
        }
    long long sum = 0;
    for (auto i : need_check)
        if (check(i))
            sum += i;
    cout << sum << endl;
    return 0;
}
